### ISSUE TITLE
download upload

### ISSUE THREAD

**Issue by robotamer** [minikomi/Bootstrap-Form-Builder#4]
I was just  gone write something like it as cli script, when I thought hmm maybe some has done that.

This is even better, thank you.

 Nice would be if we could download and upload let's say a json file rather then having to rebuild the whole thing if we need to modify something.

**Comment by minikomi** [comment]
Hmm. As it stands, that would be a bit tricky to implement. I think it would be better to consider in a future version.

**Comment by craighooghiem** [comment]
I personally rebuilt this entire thing with this method in mind. I can now re-populate the form based on data stored somewhere (file, database, etc) and it works again.

There are a couple things that need to happen though - the whole system has to work as an "instance" so that you can apply the javascript after you append the contents, otherwise the form builder won't work when you reload the existing data. I also had to do this so I can have multiple form builders on one page of my CMS applications.

Unfortunately, I'm not sure how I can re-package what I've done to share it back, since it's 100% deviated from what you have here and tied to a CMS database. The point I'm trying to make is that it's quite possible, just needs a tweak to re-load the data into the form builder and it should work.

**Comment by bluee** [comment]
It would be really nice if you could save and load a form in JSON

**Comment by minikomi** [comment]
What form would the json take? And what would you use it for?

The app is meant for quickly prototyping bootstrap forms without having to
retype all the required scaffolding markup.

On Tuesday, November 27, 2012, bluee wrote:

> It would be really nice if you could save and load a form in JSON
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/minikomi/Bootstrap-Form-Builder/issues/4#issuecomment-10730354.

## 

----=^.^=---

**Comment by bluee** [comment]
Hi, thanks for your quick response!

I would use it for creating dynamic forms, so that users could create forms on the fly.
Something like https://github.com/joshfire/jsonform except yours is way much cooler :)

**Comment by craighooghiem** [comment]
This came up in discussion earlier.. I think there's a StackOverflow question on the matter also.
What you built this for and what it can be used for are two very different things. I am using it as part of a CMS that allows the user to create (and later on edit) forms for the front-end of their website. 

So, I adjusted the generate functions so that the database now stores the "post generate" code that is clean and useful on the front-end of the website, but I also stored the un-edited code in the database so I can re-load a form from the database, populate that side of the form with the old markup and then make any changes necessary.

Worked out quite well when it was all said and done.
